<?xml version="1.0" encoding="utf-8"?>	
	
<launch>
  <!-- ICP configuration file -->
  <arg name="icp_configuration_file" default="$(find graph_image_laser)/cfg/icp_dynamic_indoor.yaml"/>
  <!-- Input filters file -->
  <arg name="icp_input_filters_file" default="$(find graph_image_laser)/cfg/input_filters_indoor.yaml"/>

  <arg name="config_path" default = "$(find graph_image_laser)/config/camera_config.yaml" />
  <arg name="pattern_file" default = "/home/ren/catkin_ws/src/VINS-Mono/support_files/brief_pattern.yml" />
  <arg name="voc_file" default = "/home/ren/catkin_ws/src/VINS-Mono/support_files/brief_k10L6.bin" />

  <!-- Enable callgrind profiling -->
  <arg name="enable_callgrind" default="false"/>
  <arg name="callgrind_ouput_path" value="$(find graph_image_laser)/profiling/segmatch-callgrind.out"/>
  <arg name="launch_prefix" value="valgrind --tool=callgrind --callgrind-out-file=$(arg callgrind_ouput_path) --instr-atstart=no" if="$(arg enable_callgrind)"/> 
  <arg name="launch_prefix" value="" unless="$(arg enable_callgrind)"/>
  	
  <!-- graph_image_laser node -->
  <node name="graph_image_laser" pkg="graph_image_laser" type="graph_image_laser_node" output="screen" respawn="true" launch-prefix="$(arg launch_prefix)">
    <param name="icp_configuration_file" value="$(arg icp_configuration_file)"/>
    <param name="icp_input_filters_file" value="$(arg icp_input_filters_file)"/>
    <param name="config_path" value="$(arg config_path)"/>
    <param name="voc_file" value="$(arg voc_file)"/>
    <param name="pattern_file" value="$(arg pattern_file)"/>
  </node>

</launch>
